doc/modules/ensemble.rst

```diff
@@ -1,534 +1,437 @@
+larger than 10,000**. The early-stopping behaviour is controlled via the
+``early_stopping``, ``scoring``, ``validation_fraction``,
+``n_iter_no_change``, and ``tol`` parameters. It is possible to early-stop
+using an arbitrary :term:`scorer`, or just the training or validation loss.
+Note that for technical reasons, using a scorer is significantly slower than
+using the loss. By default, early-stopping is performed if there are at least
+10,000 samples in the training set, using the validation loss.
 
-Like :ref:`decision trees <tree>`, forests of trees also extend to
-:ref:`multi-output problems <tree_multioutput>`  (if Y is an array
-of shape ``(n_samples, n_outputs)``).
+Missing values support
+^^^^^^^^^^^^^^^^^^^^^^
 
-Random Forests
---------------
+:class:`HistGradientBoostingClassifier` and
+:class:`HistGradientBoostingRegressor` have built-in support for missing
+values (NaNs).
 
-In random forests (see :class:`RandomForestClassifier` and
-:class:`RandomForestRegressor` classes), each tree in the ensemble is built
-from a sample drawn with replacement (i.e., a bootstrap sample) from the
-training set.
+During training, the tree grower learns at each split point whether samples
+with missing values should go to the left or right child, based on the
+potential gain. When predicting, samples with missing values are assigned to
+the left or right child consequently::
 
-Furthermore, when splitting each node during the construction of a tree, the
-best split is found either from all input features or a random subset of size
-``max_features``. (See the :ref:`parameter tuning guidelines
-<random_forest_parameters>` for more details).
+  >>> from sklearn.ensemble import HistGradientBoostingClassifier
+  >>> import numpy as np
 
-The purpose of these two sources of randomness is to decrease the variance of
-the forest estimator. Indeed, individual decision trees typically exhibit high
-variance and tend to overfit. The injected randomness in forests yield decision
-trees with somewhat decoupled prediction errors. By taking an average of those
-predictions, some errors can cancel out. Random forests achieve a reduced
-variance by combining diverse trees, sometimes at the cost of a slight increase
-in bias. In practice the variance reduction is often significant hence yielding
-an overall better model.
+  >>> X = np.array([0, 1, 2, np.nan]).reshape(-1, 1)
+  >>> y = [0, 0, 1, 1]
 
-In contrast to the original publication [B2001]_, the scikit-learn
-implementation combines classifiers by averaging their probabilistic
-prediction, instead of letting each classifier vote for a single class.
+  >>> gbdt = HistGradientBoostingClassifier(min_samples_leaf=1).fit(X, y)
+  >>> gbdt.predict(X)
+  array([0, 0, 1, 1])
 
-A competitive alternative to random forests are
-:ref:`histogram_based_gradient_boosting` (HGBT) models:
+When the missingness pattern is predictive, the splits can be done on
+whether the feature value is missing or not::
```

Comment:
```suggestion
When the missingness pattern is predictive, the splits can be performed on
whether the feature value is missing or not::
```
I feel like explicitly mentioning that this is in addition to the normal splitting rules that are evaluated would help. But I can't come up with a compact sentence/edit that does that.